src/leap/mail/mail.py

```diff
@@ -837,10 +854,16 @@ def get_collection_by_mailbox(self, name):
         """
         :rtype: MessageCollection
         """
+        collection = self._collection_mapping.get(name, None)
+        if collection:
+            return collection
```

Comment:
Yes, you're right.
In case you wonder how this could possibly work, it's because it was called in the middle of a deferred chain, so the result was the same.
i'll change this by a defer.Deferred(result) so that the return type is consistent.